src/components/Button/Button.tsx

```diff
@@ -27,150 +29,97 @@ export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
     | 'text'
   danger?: boolean
   htmlType?: 'button' | 'submit' | 'reset'
-  ref?: any
   ariaSelected?: boolean
   ariaControls?: string
   tabIndex?: 0 | -1
   role?: string
-  as?: keyof JSX.IntrinsicElements
+  containerRef?: Ref<HTMLSpanElement>
 }
+type ButtonComponent = RefForwardingComponentWithAsProps<'button', ButtonProps>
 
-interface CustomButtonProps extends React.HTMLAttributes<HTMLOrSVGElement> {}
-
-export interface RefHandle {
-  container: () => HTMLElement | null
-  button: () => HTMLButtonElement | null
-}
-
-const Button = forwardRef<RefHandle, ButtonProps>(
+const Button: ButtonComponent = forwardRef(
   (
     {
       block,
       className,
       children,
       danger,
       disabled = false,
-      onClick,
       icon,
       iconRight,
       loading = false,
       loadingCentered = false,
       shadow = true,
       size = 'tiny',
-      style,
       type = 'primary',
       htmlType,
       ariaSelected,
       ariaControls,
-      tabIndex,
-      role,
       as,
+      containerRef,
       ...props
-    }: ButtonProps,
+    },
     ref
   ) => {
-    // button ref
-    const containerRef = useRef<HTMLElement>(null)
-    const buttonRef = useRef<HTMLButtonElement>(null)
-
-    useImperativeHandle(ref, () => ({
-      container: () => {
-        return containerRef.current
-      },
-      button: () => {
-        return buttonRef.current
-      },
-    }))
-
     // styles
     const showIcon = loading || icon
 
-    let classes = [ButtonStyles['sbui-btn']]
-    let containerClasses = [ButtonStyles['sbui-btn-container']]
-
-    classes.push(ButtonStyles[`sbui-btn-${type}`])
-
-    if (block) {
-      containerClasses.push(ButtonStyles['sbui-btn--w-full'])
-      classes.push(ButtonStyles['sbui-btn--w-full'])
-    }
-
-    if (danger) {
-      classes.push(ButtonStyles['sbui-btn--danger'])
-    }
-
-    if (shadow && type !== 'link' && type !== 'text') {
-      classes.push(ButtonStyles['sbui-btn-container--shadow'])
-    }
-
-    if (size) {
-      classes.push(ButtonStyles[`sbui-btn--${size}`])
-    }
+    const classes = c8s(
```

Comment:
Hello, there is a small npm package called clsx, that can be really useful in this case. https://www.npmjs.com/package/clsx
In my opinion clsx increases the readiblity. I've used it with material UI and it's quite useful